snuba/query/mql/context_population.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
```

Comment:
Moved all of this logic to this file to simplify the parser code a bit. Also made it able to work for simple and formula queries.